Description: Use the system's double-conversion library
 There is a version of double-conversion in third-party that shouldn't be used.
Author: Faidon Liambotis <paravoid@debian.org>
Forwarded: no
Last-Update: 2014-05-26

--- /dev/null
+++ b/CMake/FindDoubleConversion.cmake
@@ -0,0 +1,22 @@
+# Finds libdouble-conversion.
+#
+# This module defines:
+# DOUBLE_CONVERSION_INCLUDE_DIR
+# DOUBLE_CONVERSION_LIBRARY
+#
+
+find_path(DOUBLE_CONVERSION_INCLUDE_DIR
+    NAMES double-conversion.h
+    PATHS /usr/include/double-conversion)
+find_library(DOUBLE_CONVERSION_LIBRARY NAMES double-conversion)
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(
+    DOUBLE_CONVERSION DEFAULT_MSG
+    DOUBLE_CONVERSION_LIBRARY DOUBLE_CONVERSION_INCLUDE_DIR)
+
+if (NOT DOUBLE_CONVERSION_FOUND)
+    message(FATAL_ERROR "Could not locate double-conversion.")
+endif (NOT DOUBLE_CONVERSION_FOUND)
+
+mark_as_advanced(DOUBLE_CONVERSION_INCLUDE_DIR DOUBLE_CONVERSION_LIBRARY)
--- a/CMake/HPHPFindLibs.cmake
+++ b/CMake/HPHPFindLibs.cmake
@@ -147,6 +147,10 @@ include_directories(${LIBSQLITE_INCLUDE_
 find_package(LZ4 REQUIRED)
 include_directories(${LZ4_INCLUDE_DIR})
 
+# libdouble-conversion
+find_package(DoubleConversion REQUIRED)
+include_directories(${DOUBLE_CONVERSION_INCLUDE_DIR})
+
 # ICU
 find_package(ICU REQUIRED)
 if (ICU_FOUND)
@@ -453,9 +457,9 @@ macro(hphp_link target)
 
   target_link_libraries(${target} ${LIBSQLITE3_LIBRARY})
   target_link_libraries(${target} ${LZ4_LIBRARY})
+  target_link_libraries(${target} ${DOUBLE_CONVERSION_LIBRARY})
 
   target_link_libraries(${target} timelib)
-  target_link_libraries(${target} double-conversion)
   target_link_libraries(${target} folly)
   target_link_libraries(${target} zip)
 
--- a/third-party/CMakeLists.txt
+++ b/third-party/CMakeLists.txt
@@ -27,7 +27,6 @@ list(APPEND THIRD_PARTY_MODULES
   libafdt
   libmbfl
   timelib
-  double-conversion
   folly)
 if(ENABLE_FASTCGI)
   list(APPEND THIRD_PARTY_MODULES ti)
--- a/CMake/HPHPSetup.cmake
+++ b/CMake/HPHPSetup.cmake
@@ -160,7 +160,6 @@ include_directories("${TP_DIR}/libafdt/s
 include_directories("${TP_DIR}/libmbfl")
 include_directories("${TP_DIR}/libmbfl/mbfl")
 include_directories("${TP_DIR}/libmbfl/filter")
-include_directories("${TP_DIR}/double-conversion/src")
 include_directories("${TP_DIR}/folly")
 include_directories(${TP_DIR})
 
